headers/extended_matrix.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "extended_types.h"
```

Comment:
Maybe `MatrixUtil.h`?

Reply:
Maybe. I didn't really bother with the names.